# key for defining mpc misc. metrics
    #float32 sqp_iter
    float32 cost
    #float32[] slacks
    #float32[] residuals
    float32 mpc_time
    float32 obsrv_time
    float32 ref_time
    float32 acados_time
    float32 qp_time
    float32 sim_time